2023-Oct-18/Dave/main.py

```diff
@@ -80,12 +84,15 @@ def get_all_legal_placements_for_word(
 
                 legal_placements.append(Placement(position, direction))
 
-    return legal_placements
+    if len(legal_placements) == 0:
+        raise NoLegalPlacementsError()
+    else:
+        return legal_placements
 
 
 def is_legal_placement(target_line: str, line_to_write: str) -> bool:
     for target_char, char_to_write in zip(target_line, line_to_write):
-        if char_to_write != target_char and target_char != " ":
+        if (char_to_write is not target_char) and (target_char != " "):
```

Comment:
A mix of ` is not` and `!=` here again.

Reply:
Ah yeah, this was because I had intended to use `is` / `is not` with everything, but received a SyntaxWarning about using it with a literal.

Turns out this is because `is` / `is not` and `==` / `!=` are not interchangeable and the former checks type as well, as far as I can tell.

Seems like the answer is to make everything `==` / `!=`.

That's enough backticks for now.